pkg/container/qemu_runner.go

```diff
@@ -717,7 +717,25 @@ func createMicroVM(ctx context.Context, cfg *Config) error {
 	// panic=-1 ensures that if the init fails, we immediately exit the machine
 	// Add default SSH keys to the VM
 	sshkey := base64.StdEncoding.EncodeToString(pubKey)
-	baseargs = append(baseargs, "-append", kernelConsole+" nomodeset random.trust_cpu=on panic=-1 sshkey="+sshkey+" melange_qemu_runner=1")
+
+	// Build kernel command line arguments
+	kernelArgs := kernelConsole + " nomodeset random.trust_cpu=on panic=-1 sshkey=" + sshkey + " melange_qemu_runner=1"
+
+	// Check for TESTING environment variable and pass it to microvm-init
+	// TESTING must be a number (or empty string)
+	if testingValue, ok := os.LookupEnv("TESTING"); ok {
+		if testingValue == "" {
+			log.Debugf("qemu: TESTING env set to empty string, passing to microvm-init via kernel cmdline")
+			kernelArgs += " melange.testing="
```

Comment:
What do you expect to use the empty space setting for? Is that expected to map to/be treated by microvm-init as zero or non-zero?

Personally, I'd stick with environment variable unset, 0, and non-zero numeric as possible values. Trying to keep the logic around this as simple as possible is probably for the best.

(I'm trying hard not to bikeshed on the variable name(s).)

Reply:
This was more input validation than a valid use case but yes I can reduce this down to set number or not set